numpy/_core/tests/test_defchararray.py

```diff
@@ -679,8 +679,8 @@ def test_mul(self):
             assert_array_equal(Ar, (self.A * r))
 
         for ob in [object(), 'qrs']:
-            with assert_raises_regex(ValueError,
```

Comment:
sorry if this was here in an earlier version and I missed it, but does this code predate our work on the string ufuncs? If so this is a backward incompatible change and we probably need to catch the ValueError and re-raise as a TypeError, specifically in the chararray implementation (blegh).

If that's really annoying to deal with then I think we need to note this behavior change in the release notes.